api/spec/controllers/mno_enterprise/jpi/v1/admin/impac/kpis_controller_spec.rb

```diff
@@ -42,34 +42,46 @@ module MnoEnterprise
       end
 
       subject { post :create, dashboard_template_id: template.id, kpi: kpi_params }
-      
-      before do
-        api_stub_for(
-          get: "/dashboards/#{template.id}",
-          params: { filter: { 'dashboard_type' => 'template' } },
-          response: from_api(template)
-        )
-        api_stub_for(
-          post: "dashboards/#{template.id}/kpis",
-          response: from_api(kpi)
-        )
-        # Why is Her doing a GET /kpis after doing a POST /kpis?
-        api_stub_for(
-          get: "dashboards/#{template.id}/kpis",
-          response: from_api([kpi])
-        )
-      end
-
-      it_behaves_like "a jpi v1 admin action"
 
-      it 'returns a kpi' do
-        subject
-        expect(JSON.parse(response.body)).to eq(hash_for_kpi)
+      context 'when the template exists' do
+        before do
+          api_stub_for(
+            get: "/dashboards/#{template.id}",
+            params: { filter: { 'dashboard_type' => 'template' } },
+            response: from_api(template)
+          )
+          api_stub_for(
+            post: "dashboards/#{template.id}/kpis",
+            response: from_api(kpi)
+          )
+          # Why is Her doing a GET /kpis after doing a POST /kpis?
```

Comment:
This is odd.

Reply:
Yeah that's copy paste from the other spec 😂